guide/groovy.md

```diff
@@ -114,14 +114,15 @@ The following API is available:
 
 | Method | Description |
 | -------|:------------|
-| [File](https://docs.oracle.com/javase/7/docs/api/java/io/File.html) getBaseDir() | Get the working directory (the execution root). |
-| [InvocationStage](https://github.com/release-engineering/pom-manipulation-ext/blob/master/core/src/main/java/org/commonjava/maven/ext/core/groovy/InvocationStage.java) getInvocationStage() | Return the current stage of the groovy manipulation. |
+| [File](https://docs.oracle.com/javase/7/docs/api/java/io/File.html) getBaseDir() | Get the working directory (the execution root) |
+| [InvocationStage](https://github.com/release-engineering/pom-manipulation-ext/blob/master/core/src/main/java/org/commonjava/maven/ext/core/groovy/InvocationStage.java) getInvocationStage() | Return the current stage of the groovy manipulation |
 | [ProjectVersionRef](https://github.com/Commonjava/atlas/blob/master/identities/src/main/java/org/commonjava/atlas/maven/ident/ref/ProjectVersionRef.java) getGAV() | Get the GAV of the current project |
 | [ModelIO](https://github.com/release-engineering/pom-manipulation-ext/blob/master/io/src/main/java/org/commonjava/maven/ext/io/ModelIO.java) getModelIO() | Return a ModelIO instance for artifact resolving |
 | [Project](https://github.com/release-engineering/pom-manipulation-ext/blob/master/common/src/main/java/org/commonjava/maven/ext/common/model/Project.java) getProject() | Return the current Project |
-| [ArrayList](https://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html)<[ProjectVersionRef](https://github.com/Commonjava/atlas/blob/master/identities/src/main/java/org/commonjava/atlas/maven/ident/ref/ProjectVersionRef.java)> getProjects() | Returns the entire collection of Projects |
+| [List](https://docs.oracle.com/javase/7/docs/api/java/util/List.html)<[Project](https://github.com/release-engineering/pom-manipulation-ext/blob/master/common/src/main/java/org/commonjava/maven/ext/common/model/Project.java)> getProjects() | Returns the entire collection of Projects |
 | [MavenSessionHandler](https://github.com/release-engineering/pom-manipulation-ext/blob/master/common/src/main/java/org/commonjava/maven/ext/common/session/MavenSessionHandler.java) getSession() | Return the current session handler |
-| [Properties](https://docs.oracle.com/javase/7/docs/api/java/util/Properties.html) getUserProperties() | Get the user properties. |
+| [Properties](https://docs.oracle.com/javase/7/docs/api/java/util/Properties.html) getUserProperties() | Get the user properties |
+| [Translator](https://github.com/release-engineering/pom-manipulation-ext/blob/master/io/src/main/java/org/commonjava/maven/ext/io/rest/Translator.java) getRESTAPI() throws [ManipulationException](https://github.com/release-engineering/pom-manipulation-ext/blob/master/common/src/main/java/org/commonjava/maven/ext/common/ManipulationException.java) | Gets a configured VersionTranslator to make REST calls to DA |
```

Comment:
I think this is the wrong place as it wasn't part of the original API so needs moving down to around line 170? (need to check when it was available)

Reply:
You mean move it to a section without the table? I originally had something there and deleted it. Can't we somehow make a note, like a footnote, in the table? I think it looks better in one place like this.

To summarize: I added `getRESTAPI()` to table and fixed the `getProjects()` return type. I did not find any other missing/incorrect APIs. I also removed some of the periods from the descriptions so that they now consistently don't have periods rather than a mix of both. You could pick one or the other. There's still an inconsistency in the Javadoc itself with no consistent standard used there, but I think the table looks better now. The table/javadoc still mixes verbs, e.g. _get_ vs. _gets_, but seems to use _return_ over _returns_ consistently there. Looking at Java's own docs, they seem to mix also depending on who did the writing.

Reply:
There are *multiple* tables depending upon the version of pme. Look at the rendered version https://release-engineering.github.io/pom-manipulation-ext/guide/groovy.html#api .... the caveat with my comment is that I can't remember without checking which version that function call became available in?

Reply:
> Can't we somehow make a note, like a footnote, in the table?

Or a "Since" column to the table?

Reply:
Well possibly but that's not how it's been done currently and would be separate IMHO to this work

Reply:
OK, I moved it. It's in its own section for version 4.3 (that is the next version, right?)